### ISSUE TITLE
empty cells

### ISSUE THREAD

**Issue by dovys** [ducksboard/gridster.js#66]
Hi, on one of my projects I need to leave empty spaces between some of the blocks. It would be nice to have a setting, where if it's set, gridster would not snap blocks one to each other.

**Comment by nwittstruck** [comment]
Yes, that would be great. I tried adding invisible placeholders as a work around, but that is kind of messy.

**Comment by facultymatt** [comment]
+1

**Comment by dustmoo** [comment]
You can do this on my fork by using the static widgets:

http://jsfiddle.net/XUsEU/

Grab the fork here:

https://github.com/dustmoo/gridster.js

The only caveat is that you have to disable shifting widgets down for it to work, so it may not be appropriate for all applications.

**Comment by facultymatt** [comment]
Looks good @dustmoo. This won't work in my use case because the static widget needs to be locked in place. I wasn't able to move the static widget. Ideally we would not even need static widgets :) and the user could move any widget to any location on the grid.

**Comment by dustmoo** [comment]
@facultymatt Sure, I was demonstrating how you could use a static widget as a "Space" as it were. But perhaps what you are thinking of is different. No worries. I wish I had more time to devote to making this library more awesome. :)

Thanks for your feedback.

**Comment by harshmsp** [comment]
Hi @dustmoo 
Is there any way to find out number of empty cells ( block) available from the panel.
I want to store number of cells and rows available from the panel.
Can you please help me.

**Comment by dustmoo** [comment]
I think you are looking for this method:

Line 604 in src/jquery.gridster.js

/**

```
* Move down widgets in cells represented by the arguments col, row, size_x,


* size_y


*


* @method empty_cells


* @param {Number} col The column where the group of cells begin.


* @param {Number} row The row where the group of cells begin.


* @param {Number} size_x The number of columns that the group of cells


* occupy.


* @param {Number} size_y The number of rows that the group of cells


* occupy.


* @param {HTMLElement} $exclude Exclude widgets from being moved.


* @return {Class} Returns the instance of the Gridster Class.


*/


fn.empty_cells = function(col, row, size_x, size_y, $exclude)
```
